pkg/data/management/role_data.go

```diff
@@ -618,6 +628,28 @@ func BootstrapAdmin(management *wrangler.Context) (string, error) {
 				logrus.Info("Created default admin user and binding")
 			}
 		}
+	} else {
+		// if we have at least one user, grab the admin user to ensure that it has the Principal label
+		var adminUser *v3.User
+		for _, user := range users.Items {
+			// we could re-list using a label selector, but since we already have a full listing, just use that instead
+			if value, ok := user.Labels[defaultAdminLabelKey]; ok {
+				if value == defaultAdminLabelValue {
+					adminUser = &user
+					break
+				}
+			}
+		}
+		if adminUser == nil {
+			logrus.Errorf("Have more than one user, but unable to locate admin user to adjust labels")
+		} else {
+			admin, err := addPrincipalLabelToAdmin(adminUser, management)
+			if err != nil {
+				logrus.Errorf("Unable to add label to adminUser %s", err.Error())
```

Comment:
Would we want to also return the error here if unable to add label to Admin User, just like in the case above when there are no users?

Reply:
I've actually changed the above to function the same was as this does. My reasoning is that if we fail to label the user for some reason, we would want to just create the admin in this diminished state. The admin is a bit too vital to not be created because of this, especially when we have an ensure admin script to fix manually if need be. Open to hearing your thoughts though.

Reply:
This is called during startup, so if an error is returned here, it will cause the Rancher service to fail to start. IMO that is the correct behavior. As you said, the admin user is too vital. It's best to fail early and let the user handle it.

Reply:
So I've changed this to warnf for now, but I'm willing to revisit this (and the below suggestion by @cmurphy) if there's significant disagreement.

I'm thinking that while the label is not a good thing to have missing, the system is technically functional without it. Therefore, it's best to warn the user about the failure and continue to move on. I'm also not sure that erroring out here would be helpful - an error in this section would be much more indicative of a failure on our side than anything that the user can actually fix. 

Though in light of the issue which spawned this PR/discussion I'm inclined to agree now that this value is critical, so I'm open to disagreement.